gradle-plugin/src/main/kotlin/com/google/devtools/ksp/gradle/KspSubplugin.kt

```diff
@@ -477,7 +478,7 @@ abstract class KspTaskJvm : KotlinCompile(KotlinJvmOptionsImpl()), KspTask {
         )
 
         isIntermoduleIncremental = project.findProperty("ksp.incremental.intermodule")?.toString()?.toBoolean() ?: true
-        if (isIncremental && isIntermoduleIncremental) {
+        if (isKspIncremental && isIntermoduleIncremental) {
```

Comment:
it looks to me that `isIntermoduleIncremental` is always set to be overridden by `isKspIncremental`, can we merge the check into where we assign the value of `isIntermoduleIncremental`?